lib/Robocluster.cpp

```diff
@@ -105,11 +105,34 @@ void run_event_loop(){
     }
 }
 
-/*
-void publish(String event, JsonObject& data, Port port){
 
+void publish(String event, JsonObject& data){
+    string slash = '/' //need to use for topic field
+    StaticJsonBuffer<200> jsonBuffer_publish;
+    JsonObject& root = jsonBuffer_publish.createObject();
+
+    root["source"] = g_device_name;
+    root["type"] = "publish";
+    JsonArray& data_nested = root.createNestedObject("data");
+    data_nested["topic"] = g_device_name + slash + event; 
+    data_nested["data"] = data;
+
+    root.printTo(Serial); //pass into serial
+}
+
+void heartbeat(){
+    StaticJsonBuffer<200> jsonBuffer_heartbeat;
+    JsonObject& root = jsonBuffer_heartbeat.createObject();
+
+    root["source"] = g_device_name;
+    root["type"] = "heartbeat";
+    JsonArray& data_nested = root.createNestedObject("data");
+    data_nested["source"] = g_device_name; 
+    data_nested["listen"] = "/dev/ttyACM0"; 
```

Comment:
I'm not really sure what we should do for the 'listen' field in the case of serial devices.  Maybe set it to 'none' for now, and maybe on the python end we will fill it in with the correct port.